repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -57,6 +58,57 @@ object Storage{
       }
     }
 
+    def compileCacheSave(tag: String, classFiles: Traversable[(String, Array[Byte])], imports: Seq[ImportData]): Unit = {
+      val cacheDir = new File(dir + s"/compileCache/$tag")
+      if(!cacheDir.exists){
+        cacheDir.mkdirs()
+        val metadata = upickle.default.write(imports)
+        val metadataFw = new FileWriter(cacheDir + "/metadata.json")
+        metadataFw.write(metadata)
+        metadataFw.flush()
+        classFiles.foreach{ case (name, bytes) =>
+          val fos = new java.io.FileOutputStream(cacheDir + s"/$name.class")
+          fos.write(bytes)
+          fos.flush()
+        }
+      }
+    }
+
+    def compileCacheLoad(tag: String): Option[(Traversable[(String, Array[Byte])], Seq[ImportData])] = {
+      val cacheDir = new File(dir + s"/compileCache/$tag")
+      if(!cacheDir.exists) None
+      else for{
+        metadataJson <- try{
+          Some(io.Source.fromFile(cacheDir + "/metadata.json").mkString)
+        } catch {
+          case e: java.io.FileNotFoundException => None
+        }
+        metadata <- try{
+          Some(upickle.default.read[Seq[ImportData]](metadataJson))
+        } catch {
+          case e: Exception => None
+        }
+        classFiles <- loadClassFiles(cacheDir)
+      } yield (classFiles, metadata)
+    }
+
+    def loadClassFiles(cacheDir: File): Option[Traversable[(String, Array[Byte])]] = {
+      val classFiles = cacheDir.listFiles().filter(_.getName.endsWith(".class"))
+      try{
+        val data = classFiles.map{ file =>
+          val fis = new java.io.FileInputStream(file)
```

Comment:
Same as output, I think we've reached the point where this should probably go in a helper function and we should refactor `ivyCache`/`loadPredef` to use that and call `new String(...)` on the byte array